packages/server/tests/acceptance/hbarLimiter.spec.ts

```diff
@@ -162,6 +162,28 @@ describe('@hbarlimiter HBAR Limiter Acceptance Tests', function () {
         verifyRemainingLimit(expectedCost, remainingHbarsBefore, remainingHbarsAfter);
       });
 
+      it('Should preemtively check the rate limit before submitting EthereumTransaction', async function () {
+        const remainingHbarsBefore = Number(await metrics.get(testConstants.METRICS.REMAINING_HBAR_LIMIT));
+
+        process.env.HBAR_RATE_LIMIT_PREEMTIVE_CHECK = 'true';
+        process.env.HOT_FIX_FILE_APPEND_FEE = (remainingHbarsBefore - 100000000).toString();
+
+        try {
+          const largeContract = await Utils.deployContract(
+            largeContractJson.abi,
+            largeContractJson.bytecode,
+            accounts[0].wallet,
+          );
+          await largeContract.waitForDeployment();
+
+          expect(true).to.be.false;
```

Comment:
`chai` provides methods to test whether a function call throws an exception, see https://www.chaijs.com/api/bdd/#method_throw. Maybe we should use that instead of a `expect(true).to.be.false`.